Failing test: Rules Management - Prebuilt Rules (Common) Integration Tests - ESS Basic License.x-pack/solutions/security/test/security_solution_api_integration/test_suites/detections_response/rules_management/prebuilt_rules/common/install_prebuilt_rules/review_installation·ts - Rules Management - Prebuilt Rules (Common tests) @ess @serverless @skipInServerlessMKI Review installation using mocked prebuilt rule assets Pagination returns correct rules for a page specified in the request

A test failed on a tracked branch

```
JestAssertionError: expect(received).toMatchObject(expected)

- Expected  -  4
+ Received  + 92

  Object {
    "rules": Array [
-     ObjectContaining {
-       "rule_id": "rule-1",
-     },
-     ObjectContaining {
+     Object {
+       "actions": Array [],
+       "author": Array [],
+       "created_at": "2026-01-19T11:40:01.022Z",
+       "created_by": "",
+       "description": "some description",
+       "enabled": false,
+       "exceptions_list": Array [],
+       "false_positives": Array [],
+       "from": "now-6m",
+       "id": "02fd2282-860e-4768-9aec-2b8ae69f30d0",
+       "immutable": true,
+       "index": Array [
+         "index-1",
+         "index-2",
+       ],
+       "interval": "5m",
+       "language": "kuery",
+       "license": "Elastic License v2",
+       "max_signals": 100,
+       "name": "Query with a rule id",
+       "output_index": "",
+       "query": "user.name: root or user.name: admin",
+       "references": Array [],
+       "related_integrations": Array [],
+       "required_fields": Array [],
+       "revision": 1,
+       "risk_score": 55,
+       "risk_score_mapping": Array [],
        "rule_id": "rule-2",
+       "rule_source": Object {
+         "customized_fields": Array [],
+         "has_base_version": true,
+         "is_customized": false,
+         "type": "external",
+       },
+       "setup": "",
+       "severity": "high",
+       "severity_mapping": Array [],
+       "tags": Array [],
+       "threat": Array [],
+       "to": "now",
+       "type": "query",
+       "updated_at": "2026-01-19T11:40:01.022Z",
+       "updated_by": "",
+       "version": 1,
+     },
+     Object {
+       "actions": Array [],
+       "author": Array [],
+       "created_at": "2026-01-19T11:40:01.022Z",
+       "created_by": "",
+       "description": "some description",
+       "enabled": false,
+       "exceptions_list": Array [],
+       "false_positives": Array [],
+       "from": "now-6m",
+       "id": "410ee426-f4f4-4d01-bd6a-3281f4f69756",
+       "immutable": true,
+       "index": Array [
+         "index-1",
+         "index-2",
+       ],
+       "interval": "5m",
+       "language": "kuery",
+       "license": "Elastic License v2",
+       "max_signals": 100,
+       "name": "Query with a rule id",
+       "output_index": "",
+       "query": "user.name: root or user.name: admin",
+       "references": Array [],
+       "related_integrations": Array [],
+       "required_fields": Array [],
+       "revision": 1,
+       "risk_score": 55,
+       "risk_score_mapping": Array [],
+       "rule_id": "rule-3",
+       "rule_source": Object {
+         "customized_fields": Array [],
+         "has_base_version": true,
+         "is_customized": false,
+         "type": "external",
+       },
+       "setup": "",
+       "severity": "high",
+       "severity_mapping": Array [],
+       "tags": Array [],
+       "threat": Array [],
+       "to": "now",
+       "type": "query",
+       "updated_at": "2026-01-19T11:40:01.022Z",
+       "updated_by": "",
+       "version": 1,
      },
    ],
  }
    at Context.<anonymous> (review_installation.ts:198:31)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at Object.apply (wrap_function.js:74:16) {
  matcherResult: {
    message: '\x1B[2mexpect(\x1B[22m\x1B[31mreceived\x1B[39m\x1B[2m).\x1B[22mtoMatchObject\x1B[2m(\x1B[22m\x1B[32mexpected\x1B[39m\x1B[2m)\x1B[22m\n' +
      '\n' +
      '\x1B[32m- Expected  -  4\x1B[39m\n' +
      '\x1B[31m+ Received  + 92\x1B[39m\n' +
      '\n' +
      '\x1B[2m  Object {\x1B[22m\n' +
      '\x1B[2m    "rules": Array [\x1B[22m\n' +
      '\x1B[32m-     ObjectContaining {\x1B[39m\n' +
      '\x1B[32m-       "rule_id": "rule-1",\x1B[39m\n' +
      '\x1B[32m-     },\x1B[39m\n' +
      '\x1B[32m-     ObjectContaining {\x1B[39m\n' +
      '\x1B[31m+     Object {\x1B[39m\n' +
      '\x1B[31m+       "actions": Array [],\x1B[39m\n' +
      '\x1B[31m+       "author": Array [],\x1B[39m\n' +
      '\x1B[31m+       "created_at": "2026-01-19T11:40:01.022Z",\x1B[39m\n' +
      '\x1B[31m+       "created_by": "",\x1B[39m\n' +
      '\x1B[31m+       "description": "some description",\x1B[39m\n' +
      '\x1B[31m+       "enabled": false,\x1B[39m\n' +
      '\x1B[31m+       "exceptions_list": Array [],\x1B[39m\n' +
      '\x1B[31m+       "false_positives": Array [],\x1B[39m\n' +
      '\x1B[31m+       "from": "now-6m",\x1B[39m\n' +
      '\x1B[31m+       "id": "02fd2282-860e-4768-9aec-2b8ae69f30d0",\x1B[39m\n' +
      '\x1B[31m+       "immutable": true,\x1B[39m\n' +
      '\x1B[31m+       "index": Array [\x1B[39m\n' +
      '\x1B[31m+         "index-1",\x1B[39m\n' +
      '\x1B[31m+         "index-2",\x1B[39m\n' +
      '\x1B[31m+       ],\x1B[39m\n' +
      '\x1B[31m+       "interval": "5m",\x1B[39m\n' +
      '\x1B[31m+       "language": "kuery",\x1B[39m\n' +
      '\x1B[31m+       "license": "Elastic License v2",\x1B[39m\n' +
      '\x1B[31m+       "max_signals": 100,\x1B[39m\n' +
      '\x1B[31m+       "name": "Query with a rule id",\x1B[39m\n' +
      '\x1B[31m+       "output_index": "",\x1B[39m\n' +
      '\x1B[31m+       "query": "user.name: root or user.name: admin",\x1B[39m\n' +
      '\x1B[31m+       "references": Array [],\x1B[39m\n' +
      '\x1B[31m+       "related_integrations": Array [],\x1B[39m\n' +
      '\x1B[31m+       "required_fields": Array [],\x1B[39m\n' +
      '\x1B[31m+       "revision": 1,\x1B[39m\n' +
      '\x1B[31m+       "risk_score": 55,\x1B[39m\n' +
      '\x1B[31m+       "risk_score_mapping": Array [],\x1B[39m\n' +
      '\x1B[2m        "rule_id": "rule-2",\x1B[22m\n' +
      '\x1B[31m+       "rule_source": Object {\x1B[39m\n' +
      '\x1B[31m+         "customized_fields": Array [],\x1B[39m\n' +
      '\x1B[31m+         "has_base_version": true,\x1B[39m\n' +
      '\x1B[31m+         "is_customized": false,\x1B[39m\n' +
      '\x1B[31m+         "type": "external",\x1B[39m\n' +
      '\x1B[31m+       },\x1B[39m\n' +
      '\x1B[31m+       "setup": "",\x1B[39m\n' +
      '\x1B[31m+       "severity": "high",\x1B[39m\n' +
      '\x1B[31m+       "severity_mapping": Array [],\x1B[39m\n' +
      '\x1B[31m+       "tags": Array [],\x1B[39m\n' +
      '\x1B[31m+       "threat": Array [],\x1B[39m\n' +
      '\x1B[31m+       "to": "now",\x1B[39m\n' +
      '\x1B[31m+       "type": "query",\x1B[39m\n' +
      '\x1B[31m+       "updated_at": "2026-01-19T11:40:01.022Z",\x1B[39m\n' +
      '\x1B[31m+       "updated_by": "",\x1B[39m\n' +
      '\x1B[31m+       "version": 1,\x1B[39m\n' +
      '\x1B[31m+     },\x1B[39m\n' +
      '\x1B[31m+     Object {\x1B[39m\n' +
      '\x1B[31m+       "actions": Array [],\x1B[39m\n' +
      '\x1B[31m+       "author": Array [],\x1B[39m\n' +
      '\x1B[31m+       "created_at": "2026-01-19T11:40:01.022Z",\x1B[39m\n' +
      '\x1B[31m+       "created_by": "",\x1B[39m\n' +
      '\x1B[31m+       "description": "some description",\x1B[39m\n' +
      '\x1B[31m+       "enabled": false,\x1B[39m\n' +
      '\x1B[31m+       "exceptions_list": Array [],\x1B[39m\n' +
      '\x1B[31m+       "false_positives": Array [],\x1B[39m\n' +
      '\x1B[31m+       "from": "now-6m",\x1B[39m\n' +
      '\x1B[31m+       "id": "410ee426-f4f4-4d01-bd6a-3281f4f69756",\x1B[39m\n' +
      '\x1B[31m+       "immutable": true,\x1B[39m\n' +
      '\x1B[31m+       "index": Array [\x1B[39m\n' +
      '\x1B[31m+         "index-1",\x1B[39m\n' +
      '\x1B[31m+         "index-2",\x1B[39m\n' +
      '\x1B[31m+       ],\x1B[39m\n' +
      '\x1B[31m+       "interval": "5m",\x1B[39m\n' +
      '\x1B[31m+       "language": "kuery",\x1B[39m\n' +
      '\x1B[31m+       "license": "Elastic License v2",\x1B[39m\n' +
      '\x1B[31m+       "max_signals": 100,\x1B[39m\n' +
      '\x1B[31m+       "name": "Query with a rule id",\x1B[39m\n' +
      '\x1B[31m+       "output_index": "",\x1B[39m\n' +
      '\x1B[31m+       "query": "user.name: root or user.name: admin",\x1B[39m\n' +
      '\x1B[31m+       "references": Array [],\x1B[39m\n' +
      '\x1B[31m+       "related_integrations": Array [],\x1B[39m\n' +
      '\x1B[31m+       "required_fields": Array [],\x1B[39m\n' +
      '\x1B[31m+       "revision": 1,\x
[report_failure] output truncated to 8192 characters
```

First failure: [kibana-on-merge - main](https://buildkite.com/elastic/kibana-on-merge/builds/85915#019bd5ee-ccbd-48e1-9ea4-34021a0ff7c3)

<!-- kibanaCiData = {"failed-test":{"test.class":"Rules Management - Prebuilt Rules (Common) Integration Tests - ESS Basic License.x-pack/solutions/security/test/security_solution_api_integration/test_suites/detections_response/rules_management/prebuilt_rules/common/install_prebuilt_rules/review_installation·ts","test.name":"Rules Management - Prebuilt Rules (Common tests) @ess @serverless @skipInServerlessMKI Review installation using mocked prebuilt rule assets Pagination returns correct rules for a page specified in the request","test.failCount":13}} -->

--- a/x-pack/solutions/security/test/security_solution_api_integration/test_suites/detections_response/rules_management/prebuilt_rules/common/install_prebuilt_rules/review_installation.ts
+++ b/x-pack/solutions/security/test/security_solution_api_integration/test_suites/detections_response/rules_management/prebuilt_rules/common/install_prebuilt_rules/review_installation.ts
@@ -142,19 +142,26 @@ export default ({ getService }: FtrProviderContext): void => {
         ];
         await createPrebuiltRuleAssetSavedObjects(es, ruleAssets);
 
-        const response = await reviewPrebuiltRulesToInstall(supertest, {
+        const page1Response = await reviewPrebuiltRulesToInstall(supertest, {
           page: 1,
           per_page: 2,
         });
-        expect(response.rules.length).toEqual(2);
+        expect(page1Response.rules.length).toEqual(2);
+
+        const page2Response = await reviewPrebuiltRulesToInstall(supertest, {
+          page: 2,
+          per_page: 2,
+        });
+
+        expect(page2Response.rules.length).toEqual(1);
       });
 
       it('returns correct rules for a page specified in the request', async () => {
         const ruleAssets = [
-          createRuleAssetSavedObject({ rule_id: 'rule-1' }),
-          createRuleAssetSavedObject({ rule_id: 'rule-2' }),
-          createRuleAssetSavedObject({ rule_id: 'rule-3' }),
-          createRuleAssetSavedObject({ rule_id: 'rule-4' }),
+          createRuleAssetSavedObject({ name: 'Rule 1', rule_id: 'rule-1' }),
+          createRuleAssetSavedObject({ name: 'Rule 2', rule_id: 'rule-2' }),
+          createRuleAssetSavedObject({ name: 'Rule 3', rule_id: 'rule-3' }),
+          createRuleAssetSavedObject({ name: 'Rule 4', rule_id: 'rule-4' }),
         ];
 
         await createPrebuiltRuleAssetSavedObjects(es, ruleAssets);
@@ -162,15 +169,14 @@ export default ({ getService }: FtrProviderContext): void => {
         const page1Response = await reviewPrebuiltRulesToInstall(supertest, {
           page: 1,
           per_page: 2,
+          sort: [{ field: 'name', order: 'asc' }],
         });
 
         expect(page1Response.rules).toHaveLength(2);
-        expect(page1Response.rules).toEqual(
-          expect.arrayContaining([
-            expect.objectContaining({ rule_id: 'rule-1' }),
-            expect.objectContaining({ rule_id: 'rule-2' }),
-          ])
-        );
+        expect(page1Response.rules).toEqual([
+          expect.objectContaining({ name: 'Rule 1' }),
+          expect.objectContaining({ name: 'Rule 2' }),
+        ]);
 
         const page2Response = await reviewPrebuiltRulesToInstall(supertest, {
           page: 2,
@@ -178,12 +184,10 @@ export default ({ getService }: FtrProviderContext): void => {
         });
 
         expect(page2Response.rules).toHaveLength(2);
-        expect(page2Response.rules).toEqual(
-          expect.arrayContaining([
-            expect.objectContaining({ rule_id: 'rule-3' }),
-            expect.objectContaining({ rule_id: 'rule-4' }),
-          ])
-        );
+        expect(page2Response.rules).toEqual([
+          expect.objectContaining({ name: 'Rule 3' }),
+          expect.objectContaining({ name: 'Rule 4' }),
+        ]);
       });
 
       it('returns correct number of rules for the last page', async () => {

